### PR TITLE
Make UnityLauncherAPI work

### LINKED ISSUE #298: Unread Notification Badge [Linux]
Cross-platform duplicate of [this Win10 feature request](https://github.com/Vencord/Vesktop/issues/20).

Feature looks like this:
notif_badge.

Note: Vanilla discord has a missing dependency in libunity which is responsible for notification badges, as referenced [in this thread](<https://github.com/flathub/com.discordapp.Discord/issues/228>).
As shown above, the feature works on vanilla discord after installing libunity (or on distros that come with it), but not on Vencord.

### PR DESCRIPTION
**Pull request by JakobDev**
Fixes Vencord/Vesktop#298

### CODE REVIEW

**Comment by lewisakura** [comment]
bot, build

File: dev.vencord.Vesktop.yml [hunk 1]
```diff
@@ -24,6 +24,10 @@ finish-args:
   - --filesystem=xdg-run/pipewire-0 # Pipewire interfacing
   - --filesystem=~/.steam # Needed for SteamOS integration, as the XDG OpenURL portal doesn't work properly in Game Mode. We only need ~/.steam/steam.pipe but Flatpak can't correctly mount just that: 'File "/home/deck/.steam/steam.pipe" has unsupported type 0o10000'
   - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons # This is used to show the correct cursor on Wayland
+  - --talk-name=com.canonical.Unity
+  # Electron checks if it running on Unity/KDE before using the UnityLauncherAPI, so we need to pretend we are on KDE to make it work on other desktops
+  - --env=XDG_CURRENT_SESSION=KDE
+  - --env=XDG_CURRENT_DESKTOP=KDE
```
Comment (Vendicated) [inline dev.vencord.Vesktop.yml:30]: pretending to be a different desktop environment seems wrong. why is this necessary?

Comment (JakobDev) [reply dev.vencord.Vesktop.yml:30]: [Electron does check the current desktop](https://github.com/electron/electron/blob/fb88375ab4d2161dbf7e958a2a94c7c6d97dc84c/shell/browser/linux/unity_service.cc#L64) for some reason. If we want to make this work outside KDE, we need to pretend we are on KDE.

**Comment by Vendicated** [comment]
yeah like i said this is a bad idea. for example this completely breaks opening links because xdg-open tries to use konqueror to open it. electron 32 will remove this kde check, so we should just wait for that! thanks for the contribution anyway

image
